### PR TITLE
Fixed issues when structs where used as object types.

### PR DESCRIPTION
**Pull request by michaelstaib**

### CODE REVIEW

File: src/Core/Types.Tests/Types/ObjectTypeTests.cs [hunk 1]
```diff
@@ -1462,5 +1488,15 @@ public class FooDeprecated
 
             public string Bar2() => "Foo 2: Electric foo-galoo";
         }
+
+        public struct FooStruct
+        {
+            public string GetBar() => "Baz";
```
Comment (jotatoledo) [inline src/Core/Types.Tests/Types/ObjectTypeTests.cs]: Could you also add both a field and a property to the struct definition? In order to further improve the validity of the new tests :)

Comment (michaelstaib) [reply src/Core/Types.Tests/Types/ObjectTypeTests.cs]: Sure :)

Comment (michaelstaib) [reply src/Core/Types.Tests/Types/ObjectTypeTests.cs]: Although we are not inferring fields ... fields are ignored.

Comment (michaelstaib) [reply src/Core/Types.Tests/Types/ObjectTypeTests.cs]: But I will add a test that ensures that is the case.

Comment (michaelstaib) [reply src/Core/Types.Tests/Types/ObjectTypeTests.cs]: OK, test are now improved.

**Pull request merged**